src/plugins/discourse/mirror.test.js

```diff
@@ -568,4 +654,548 @@ describe("plugins/discourse/mirror", () => {
+        repo
+          .users()
+          .slice()
+          .sort()
+      ).toEqual(["alpha", "beta", "credbot"]);
+    });
+
+    function expectLikesSorted(as, bs) {
+      const s = (ls) =>
+        sortBy(
+          ls,
+          (x) => x.username,
+          (x) => x.postId
+        );
+      expect(s(as)).toEqual(s(bs));
+    }
+
+    it("provides all the likes by users that have posted", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+
+      fetcher.addPost(1, null, "alpha");
+      fetcher.addPost(2, null, "alpha");
+      fetcher.addPost(3, null, "beta");
+      const l1 = fetcher.addLike("beta", 1, 5);
+      const l2 = fetcher.addLike("beta", 2, 6);
+      const l3 = fetcher.addLike("beta", 3, 7);
+      const l4 = fetcher.addLike("alpha", 1, 8);
+      await mirror.update(reporter);
+      expectLikesSorted(repo.likes(), [l1, l2, l3, l4]);
+
+      const l5 = fetcher.addLike("alpha", 2, 9);
+      fetcher.addPost(4, null, "credbot");
+      const l6 = fetcher.addLike("credbot", 2, 10);
+      const l7 = fetcher.addLike("beta", 4, 11);
+      await mirror.update(reporter);
+      expectLikesSorted(repo.likes(), [l1, l2, l3, l4, l5, l6, l7]);
+    });
+
+    it("doesn't find likes of users that never posted", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      fetcher.addPost(1, null);
+      fetcher.addLike("nope", 1, 1);
+      await mirror.update(reporter);
+      expect(repo.likes()).toEqual([]);
+    });
+
+    it("mirrors category definition topics only after it's interval", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+
+      fetcher.addCategory(4);
+      const topic4 = fetcher._categoryDefinitionTopic(4);
+      await mirror.update(reporter);
+      const firstTopics = repo.topics();
+
+      fetcher.addCategory(5);
+      await mirror.update(reporter);
+      const laterTopics = repo.topics();
+
+      expect(firstTopics).toEqual([topic4]);
+      expect(laterTopics).toEqual([topic4]);
+    });
+
+    it("mirrors respects recheckCategoryDefinitionsAfterMs option", async () => {
+      const {mirror, fetcher, reporter, repo} = example({
+        recheckCategoryDefinitionsAfterMs: 0,
+      });
+
+      fetcher.addCategory(4);
+      const topic4 = fetcher._categoryDefinitionTopic(4);
+      await mirror.update(reporter);
+      const firstTopics = repo.topics();
+
+      fetcher.addCategory(5);
+      const topic5 = fetcher._categoryDefinitionTopic(5);
+      await mirror.update(reporter);
+      const laterTopics = repo.topics();
+
+      expect(firstTopics).toEqual([topic4]);
+      expect(laterTopics).toEqual([topic4, topic5]);
+    });
+
+    it("mirrors respects recheckTopicsInCategories option", async () => {
+      const {mirror, fetcher, reporter, repo} = example({
+        recheckTopicsInCategories: [4],
+      });
+      const cooked = "<p>New content, without bump</p>";
+
+      // Adds a category description #4.
```

Comment:
Should we also have a category description which is *not* getting rechecked?

Reply:
Wouldn't hurt :man_shrugging:. But the expected behavior of those are tested elsewhere.
The category definition topics that aren't in this list should update when `recheckCategoryDefinitionsAfterMs` has elapsed.
The normal topics in a category should be updated when they're bumped.